core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/admin/user/AdminUserResource.scala

```diff
@@ -44,11 +46,18 @@ class AdminUserResource {
       throw new WebApplicationException("Email already exists", Response.Status.CONFLICT)
     }
     val updatedUser = userDao.fetchOneByUid(user.getUid)
+    val roleChanged = updatedUser.getRole != user.getRole
     updatedUser.setName(user.getName)
     updatedUser.setEmail(user.getEmail)
     updatedUser.setRole(user.getRole)
     updatedUser.setComment(user.getComment)
     userDao.update(updatedUser)
+
+    if (roleChanged)
+      sendEmail(
```

Comment:
Add the email validator. You can have a util component to do it.

Reply:
The validation is done on frontend.

Reply:
You can put the isValidEmail check inside the `sendEmail` function, and keep it private